### ISSUE TITLE
Delay model generation: bug in timing client results in some data points with delay:0

### ISSUE THREAD

**Issue by tcal-x** [google/xls#1002]
When evaluating the minimum delay for a particular  parameterization, `timing_characterization_client.py` performs a binary search.  Due to some bug, sometimes the search terminates with *no* successful run, and when this happens, the delay value is set to 0.  This in turn causes a bad fit curve, typically a flat line.

One case where this currently happens with sky130 is kAdd, width 40 bits.

**Comment by tcal-x** [comment]
With the kAdd / 40b example, it seems to be that the first run response fails, with the timing analysis saying that an frequency of X MHz should be achievable.   The binary search then sets the upper bound to be 1.1 * X, and the lower bound to be 0.9 * X.    But then the next run fails, and indicates that the fmax can by Y at most, where Y is less than the lower bound (0.9 * X).    The binary search only searches between its lower and upper bound, and so never "passes".

I'll update the binary search to recognize when it's stuck like this, and never return a "no result" delay.

But I also need to look into the mystery of why we get different FMax estimates of X and Y.    I had observed before that Yosys / openSTA  *don't* change their mapping and timing estimate in response to different requested target frequencies.

**Comment by tcal-x** [comment]
I found the cause of the changing FMAX from Yosys/ABC.   I had used an experimental Yosys/ABC script (not checked in) that used ABC commands `upsize -D %f` and `dnsize -D %f`.    In these commands the `-D` option provides a target clock period.   After getting the estimate from the first failure, that estimate was fed back in as the target.  This caused ABC to relax its sizing optimization, resulting in a lower FMAX.  

I am addressing this by removing the `-D` options from the script.  Removing them will cause ABC to optimize for the fastest delay.   *The `-D` option will only cause ABC to give a slower mapping, never a faster mapping.*
